virtual_rainforest/core/schema.py

```diff
@@ -134,15 +135,17 @@ def merge_schemas(schemas: dict[str, dict[str, Any]]) -> dict[str, Any]:
         # Add module name to list of required modules
         comb_schema["required"].append(this_module)
 
-    p_paths = []
-    # Recursively search for all instances of properties in the schema
-    for path, value in dpath.util.search(comb_schema, "**/properties", yielded=True):
-        # Remove final properties instance from path so that additionalProperties ends
-        # up in the right place
-        p_paths.append("" if path == "properties" else path[:-11])
-
-    # Set additional properties to false everywhere that properties are defined
-    for path in p_paths:
-        dpath.util.new(comb_schema, f"{path}/additionalProperties", False)
+    # Recursively search for all instances of properties and insert
+    # additionalProperties=false to prevent undocumented properties in schema.
+    # It does seem odd that JSONSchema has no universal setting for this.
+    property_paths = [
+        path for path, _ in dpath.search(comb_schema, "**/properties", yielded=True)
+    ]
```

Comment:
Unnecessary list creation:

```suggestion
    property_paths = (
        path for path, _ in dpath.search(comb_schema, "**/properties", yielded=True)
    )
```

Reply:
No - wait. The intention here is to turn it into a **generator**, right? Not just a tuple. 

That doesn't work because as the `property_paths` generator yields paths in the following loop,  the underlying `comb_schema` is modified, which changes the dictionary size,  and breaks the `search` iteration.

Reply:
So it does. My bad.